### DISCUSSION TITLE
useFieldArray with multiple fields, how to use the item.id as key?

### DISCUSSION THREAD

**Discussion by Geddard** [react-hook-form/react-hook-form#3983] [Q&A]
I think maybe im not understanding something here, according to the API here

![image](https://user-images.githubusercontent.com/3460028/105369330-bc133f00-5c02-11eb-8954-ca02694e1a65.png)

It says i should give each field the item.id as key, but how does that work if i have a field array with many inputs?
Example for structure:

```
{
  formData: [
    {
      firstName: '',
      lastName: ''
    },
    {
      firstName: '',
      lastName: ''
    },
    {
      firstName: '',
      lastName: ''
    },
  ]
}
```

If i follow the example, in such a case, there would be two inputs with the same key, right? 
Right now im having the issue that if i use `remove(index)`, the first time it works but the second time fields has undefined items in place of the previous list... i cant figure out why

Another thing that is not clear, is the fieldArray ready to be used with Controller?
In the API it doesn't mention anything about it but it says i "should" use `ref` with `register()`

Thanks!

**Comment by bluebill1049** [comment]
the doc means at the root level to give a key, not an input level if that's your concern?

**Reply by Geddard** [reply]
> the doc means at the root level to give a key, not an input level if that's your concern?

Ok, that's what i assumed but i was asking just in case, so what about my second concern?

>Another thing that is not clear, is the fieldArray ready to be used with Controller?
In the API it doesn't mention anything about it but it says i "should" use ref with register()

**Reply by bluebill1049** [reply]
you should check out the codesandbox example in the doc, it has both.

**Comment by Geddard** [comment]
Ok, found the trick.. i had to add `shouldUnregister: false` and everything works now xD
